src/modules/idyntree-yarp-visualizer/Visualizer.cpp

```diff
@@ -282,6 +273,108 @@ void idyntree_yarp_tools::Visualizer::updateJointValues()
+{
+
+    size_t inactivityThreshold = m_maxVizFPS/2;
+
+    for (auto& vizWrench : m_netExternalWrenchesMap)
+    {
+        if (vizWrench.second.inactivityCounter < inactivityThreshold)
+        {
+            vizWrench.second.inactivityCounter++;
+        }
+    }
+
+    if (m_connectedToWBD)
+    {
+        yarp::os::Bottle* bottle = m_netExternalWrenchesPort.read(false);
+        if (bottle)
+        {
+            for (size_t link = 0; link < bottle->size(); ++link)
+            {
+                yarp::os::Bottle* wrenchPair = bottle->get(link).asList();
+                if (!wrenchPair || wrenchPair->size() != 2)
+                {
+                    return;
+                }
+
+                std::string linkName = wrenchPair->get(0).asString();
+                yarp::os::Bottle* wrenchBottle = wrenchPair->get(1).asList();
+
+                if (!wrenchBottle || wrenchBottle->size() != 6)
+                {
+                    return;
+                }
+
+                VisualizedWrench& vizWrench = m_netExternalWrenchesMap[linkName];
+
+                vizWrench.inactivityCounter = 0;
+
+                for (size_t i = 0; i < 3; ++i)
+                {
+                    vizWrench.scaledWrench(i) = m_forceMultiplier * wrenchBottle->get(i).asDouble();
+                }
+
+                for (size_t i = 3; i < 6; ++i)
+                {
+                    vizWrench.scaledWrench(i) = m_torquesMultiplier * wrenchBottle->get(i).asDouble();
+                }
+
+                if ((vizWrench.frameIndex == iDynTree::FRAME_INVALID_INDEX) && !vizWrench.skip) //the link has never been checked in the model
+                {
+                    vizWrench.frameIndex = m_viz.modelViz("robot").model().getFrameIndex(linkName);
+                    vizWrench.skip =  vizWrench.frameIndex == iDynTree::FRAME_INVALID_INDEX;
+                }
+            }
+        }
+    }
+    else
+    {
+        for (auto& vizWrench : m_netExternalWrenchesMap)
+        {
+            vizWrench.second.scaledWrench.zero();
+        }
+    }
+
+    for (auto& vizWrench : m_netExternalWrenchesMap)
+    {
+        if (vizWrench.second.inactivityCounter >= inactivityThreshold)
+        {
+            vizWrench.second.scaledWrench.zero();
+        }
+
+        if (!vizWrench.second.skip)
+        {
+            iDynTree::Transform linkTransform = m_viz.modelViz("robot").getWorldFrameTransform(vizWrench.second.frameIndex);
+            iDynTree::Wrench scaledWrenchInWorld = linkTransform * vizWrench.second.scaledWrench;
+            iDynTree::Position originLinear = linkTransform.getPosition();
+            iDynTree::Position originAngular = linkTransform.getPosition();
+
+
+            for (size_t i = 0; i < 3; ++i)
+            {
+                originLinear(i) -= scaledWrenchInWorld.getLinearVec3()(i); //to have the arrow pointing toward the link
+                originAngular(i) -= scaledWrenchInWorld.getAngularVec3()(i);
+            }
```

Comment:
Is this logic of the location of the wrenches documented or printed somewhere? It would be useful to specify this.

Reply:
It is specified in https://github.com/robotology/whole-body-estimators/blob/master/devices/wholeBodyDynamics/README.md, but I can add a comment in there.

Reply:
Done in https://github.com/robotology/idyntree-yarp-tools/pull/8/commits/0d9b2f6bd0e8b1ff21820464658375d1b6c62e93